dcos-log/api/middleware.go

```diff
@@ -0,0 +1,210 @@
+	globalAuthVars = &authVars{}
+}
+
+// validate the token
+func validateToken(t string) (string, error) {
+	if !strings.HasPrefix(t, "token=") {
+		return t, ErrInvalidToken
+	}
+
+	// JWT structure: header.payload.signature https://jwt.io/introduction/
+	if parts := strings.Split(t, "."); len(parts) != 3 {
+		return t, ErrInvalidToken
+	}
+
+	// TODO: improve token validation
+	return t, nil
+}
+
+// getAuthOrCookieFromRequest will try to extract JWT from Authorization header first. If it is not there
+// it will try to look for a dcos-acs-auth-cookie.
+func getAuthOrCookieFromRequest(r *http.Request) (string, error) {
+	// give priority to Authorization header
+	authorizationHeader := r.Header.Get("Authorization")
+	if authorizationHeader != "" {
+		return validateToken(authorizationHeader)
+	}
+
+	// if authorization header is not set and no cookie header, return empty string
+	cookieHeader := r.Header.Get("Cookie")
+	if cookieHeader == "" {
+		return "", ErrMissingToken
+	}
+
+	cookies := strings.Split(cookieHeader, ";")
+	for _, cookie := range cookies {
+		keyValue := strings.Split(cookie, "=")
+		if keyValue[0] == authCookieName {
+			return validateToken("token=" + keyValue[1])
+		}
+	}
+
+	return "", ErrInvalidCookieHeader
+}
+
+func authMiddleware(next http.Handler, cfg *config.Config, client *http.Client, dcosUtil dcos.Util) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// JWT is required to present in a request. The middleware will extract the token and try to access
+		// sandbox with it. We authorize the request if sandbox returns 200.
+		token, err := getAuthOrCookieFromRequest(r)
+		if err != nil {
+			httpError(w, fmt.Sprintf("Token error: %s", err.Error()), http.StatusUnauthorized, r)
+			return
+		}
+
+		// frameworkID, executorID and containerID are required mux variables to authorize a request.
+		frameworkID := mux.Vars(r)["framework_id"]
+		executorID := mux.Vars(r)["executor_id"]
+		containerID := mux.Vars(r)["container_id"]
+
+		// if we ended up in this handler without required mux variables, we are doing something wrong.
+		if frameworkID == "" || executorID == "" || containerID == "" {
+			httpError(w, "Missing mux variables `frameworkID`, `executorID` or `containerID`", http.StatusBadRequest, r)
+			return
+		}
+
+		sandboxBaseURL, err := globalAuthVars.getSandboxURL(cfg, dcosUtil)
+		if err != nil {
+			httpError(w, "Unable to get sandboxBaseURL: "+err.Error(), http.StatusInternalServerError, r)
+			return
+		}
+
+		mesosID, err := globalAuthVars.getMesosID(dcosUtil)
+		if err != nil {
+			httpError(w, "Unable to get mesosID: "+err.Error(), http.StatusInternalServerError, r)
+			return
+		}
+
+		sandboxPath := fmt.Sprintf("/var/lib/mesos/slave/slaves/%s/frameworks/%s/executors/%s/runs/%s/", mesosID, frameworkID, executorID, containerID)
+		sandboxBaseURL.RawQuery = "path=" + url.QueryEscape(sandboxPath)
+
+		req, err := http.NewRequest("GET", sandboxBaseURL.String(), nil)
```

Comment:
do we care about timeouts here? or is the caller expected to have configured the http.Client appropriately?